compiler/noirc_evaluator/src/ssa/function_builder/data_bus.rs

```diff
@@ -27,15 +34,21 @@ impl DataBusBuilder {
         }
     }
 
-    /// Generates a boolean vector telling which (ssa) parameter from the given function signature
+    /// Generates a vector telling which (ssa) parameters from the given function signature
     /// are tagged with databus visibility
-    pub(crate) fn is_databus(main_signature: &FunctionSignature) -> Vec<bool> {
+    pub(crate) fn is_databus(main_signature: &FunctionSignature) -> Vec<DatabusVisibility> {
         let mut params_is_databus = Vec::new();
 
         for param in &main_signature.0 {
             let is_databus = match param.2 {
-                ast::Visibility::Public | ast::Visibility::Private => false,
-                ast::Visibility::DataBus => true,
+                ast::Visibility::Public | ast::Visibility::Private => DatabusVisibility::None,
+                ast::Visibility::DataBus(id) => {
+                    if id > 0 {
```

Comment:
Why is `id = 0` used to distinguish between `CallData` and `ReturnData` instead of an `Option` or `ast::Visibility::ReturnData`?

Reply:
No particular reason, I changed to have ReturnData visibility for better clarity.